src/main/java/fi/hsl/transitdata/cancellation/util/TimeUtils.java

```diff
@@ -31,4 +42,81 @@ public static Optional<Long> toUtcEpochMs(String localTimestamp, String zoneId)
             return Optional.empty();
         }
     }
+    
+    /**
+     * Returns dates within the given time period. For example, if time period is from 2024-01-02 to 2024-01-05, this
+     * method returns the following dates as strings: '20240102', '20240103', '20240104' and '20240105'
+     * @param validFrom alert valid from timestamp
+     * @param validTo alert valid to timestamp
+     * @return list of dates as string, each date has format 'YYYYMMDD'
+     * @exception throws RuntimeException if parameter validFrom and/or validTo is null, or if validFrom is after validTo
+     */
+    public static List<String> getDates(LocalDateTime validFrom, LocalDateTime validTo) {
+        if (validFrom == null && validTo == null) {
+            throw new RuntimeException("validFrom and validTo are null");
+        } else if (validFrom == null) {
+            throw new RuntimeException("validFrom is null");
+        } else if (validTo == null) {
+            throw new RuntimeException("validTo is null");
+        } else if (validFrom.isAfter(validTo)) {
+            throw new RuntimeException("validFrom is after validTo");
+        }
+        
+        List<String> dates = new ArrayList<>();
+        
+        String validFromAsString = getDateAsString(validFrom);
+        String validToAsString = getDateAsString(validTo);
+        
+        LocalDateTime nextDate = validFrom;
+        String nextDateAsString = validFromAsString;
+        dates.add(validFromAsString);
+        
+        while (!validToAsString.equals(nextDateAsString)) {
+            nextDate = getNextDate(nextDate);
+            nextDateAsString = getDateAsString(nextDate);
+            dates.add(nextDateAsString);
+        }
+        
+        return dates;
+    }
+    
+    // Return date as format 'YYYYMMDD', for example '20240102'
+    private static String getDateAsString(LocalDateTime someDate) {
+        return someDate.format(DATE_FORMATTER);
+    }
+    
+    private static LocalDateTime getNextDate(LocalDateTime someDate) {
+        return someDate.plusDays(1);
+    }
+    
+    /**
+     * Returns date object
+     * @param dateAsString date as format 'YYYYMMDD'
+     * @param timeAsString time as format 'HHMM'
+     * @return
+     */
+    static LocalDateTime getDate(String dateAsString, String timeAsString) {
+        String timestampAsString = dateAsString + " " + timeAsString;
+        return LocalDateTime.parse(timestampAsString, DATE_TIME_FORMATTER);
+    }
+    
+    /**
+     * Get date in String, for example '20240108'.
+     * @param serviceDay Departure date of the trip. Format: Unix timestamp (local time) in seconds.
+     * @return Date as string in format 'YYYYMMDD'
+     */
+    public static String getShortDate(Integer serviceDay) {
```

Comment:
Could be renamed `getDateAsString(Integer serviceDay)` to follow convention above?

Reply:
Method getDateAsString(LocalDateTime someDate) already exists. Perhaps new naming conventions should be invented?